### PR TITLE
Task/683  implement role crud has been created

### PR DESCRIPTION
**Pull request by jananikuruntham**

### CODE REVIEW

File: api/src/groups/groups.controller.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { AuthGuard } from '@nestjs/passport';
+
+@Controller('groups')
+export class GroupsController {
+  constructor(private groupService: GroupsService) {}
+
+  @Get()
+  @HttpCode(HttpStatus.OK)
+  @UseGuards(AuthGuard('jwt'))
+  async getUsers() {
```
Comment (vsaravanakumar) [inline api/src/groups/groups.controller.ts:28]: Change the file name

File: api/src/role/role.controller.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @Request() req,
+    @Param('id') id: number,
+  ): Promise<{ message: string; statusCode: number }> {
+    return this.roleService.update(roleDto, req.user, id);
+  }
+
+  @Get('get-active-roles')
+  @HttpCode(HttpStatus.OK)
+  @UseGuards(AuthGuard('jwt'))
+  async activeRoles() {
```
Comment (vsaravanakumar) [inline api/src/role/role.controller.ts:76]: Change the file name

File: api/src/role/role.service.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    } catch (error) {
+      return {
+        message: 'An error occurred while deleting the role.',
+        statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+        data: {},
+      };
+    }
+  }
+
+  async getActiveRoles(): Promise<Role[]> {
```
Comment (vsaravanakumar) [inline api/src/role/role.service.ts:176]: Check the function and remove if not needed

File: front/src/components/User/AddUserModal.tsx [hunk 4]
```diff
@@ -113,6 +108,18 @@ export default function AddUserModal({
     }
   };
 
+  const getroles = async () => {
```
Comment (vsaravanakumar) [inline front/src/components/User/AddUserModal.tsx:111]: Change the function name to camel case

File: front/src/components/Groups/GroupList.tsx [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+      const response = await GroupsApi.updateGroupApi(id, updatedData);
+      setUniqueAlert("");
+      if (response) {
+        if (response.statusCode === 409) {
+          setUniqueAlert(response.message);
+        } else if (response.statusCode === 200) {
+          const updatedRows = rows.map((row) =>
+            row.id === id ? { ...row, ...updatedData } : row
+          );
+          console.log("pp", response);
```
Comment (vsaravanakumar) [inline front/src/components/Groups/GroupList.tsx:226]: Remove console log

**Comment by vsaravanakumar** [comment]
1. Remove unwanted migration files
2. Change role to roles on all pages including database
3. Remove the role view page code

File: api/src/groups/groups.controller.ts (hunk 1 above)
Comment (jananikuruntham) [reply api/src/groups/groups.controller.ts:28]: file name is changed

File: api/src/role/role.controller.ts (hunk 2 above)
Comment (jananikuruntham) [reply api/src/role/role.controller.ts:76]: file name is changed

File: api/src/role/role.service.ts (hunk 3 above)
Comment (jananikuruntham) [reply api/src/role/role.service.ts:176]: function is needed

File: front/src/components/User/AddUserModal.tsx (hunk 4 above)
Comment (jananikuruntham) [reply front/src/components/User/AddUserModal.tsx]: changed function name to camel case

File: front/src/components/Groups/GroupList.tsx (hunk 5 above)
Comment (jananikuruntham) [reply front/src/components/Groups/GroupList.tsx:226]: removed console log

**Pull request merged**